services/datalad/datalad_service/common/github.py

```diff
@@ -23,17 +24,38 @@ def create_github_repo(dataset_path, dataset_id):
             'DATALAD_GITHUB_TOKEN and DATALAD_GITHUB_ORG must be defined to create remote repos')
 
 
-def github_export(dataset_path, tag):
+def github_export(dataset_id, dataset_path, tag):
     """
     Publish GitHub repo and tags.
     """
     subprocess.check_call(
         ['git', 'push', 'github', f'{tag}:refs/heads/main'], cwd=dataset_path)
+    subprocess.check_call(
+        ['git', 'push', 'github', f'{tag}:refs/heads/master'], cwd=dataset_path)
     subprocess.check_call(
         ['git', 'push', 'github', 'git-annex:refs/heads/git-annex'], cwd=dataset_path)
     # Update tags
     subprocess.check_call(
         ['git', 'push', '--tags', 'github'], cwd=dataset_path)
+    # Make sure the default branch is correct
+    description = read_dataset_description(dataset_path, tag)
+    github_set_repo_config(dataset_id, description.Name)
```

Comment:
I think this is just a dict, not an object.

```suggestion
    github_set_repo_config(dataset_id, description["Name"])
```

Reply:
Yeah, I replaced this commit once I'd tested it. I think I need to also add an exception case for when the dataset_description.json git object doesn't exist, even though that should really never happen.